### PR TITLE
fix(developer,windows): improve uploading of sourcemaps to sentry

### LINKED ISSUE #10720: chore(web): Make sure sourcemaps are available for keymanweb in the Keyman Developer Debugger
Secondary issue -- we don't have sourcemap working here, which makes it hard to trace the issue.

These sourcemaps need to be uploaded to sentry.

_Originally posted by @mcdurdin in https://github.com/keymanapp/keyman/issues/10719#issuecomment-1945377673_

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #10720.

This makes a number of fixes:
* Uploads each sourcemap path separately to try and avoid the 413 error noted in the issue above.
* Adds the `--no-dedupe` flag to avoid another 414 error from sentry.
* Starts from the parent folder, to avoid paths starting with .. (just tidies things up, no functional change)
* Removes a number of unneeded test files and debug symbols, which should reduce the size of each release's debug data upload (this is not a sourcemap change!)
* Removes the 'dist' property from sourcemap uploads, as we don't use it, and that causes sourcemaps to not match.
* Fixes an error where some [crash reports](https://keyman.sentry.io/issues/4988886047/) had the version tag `release@18.0.1.0_GIT_TAG` instead of `release@18.0.1-alpha` 
* Updates the sentry-browser SDK to 7.102.0 on the following components:
  * Keyman Developer Server
  * Keyman Developer internal web UI
  * Keyman Configuration
  * FirstVoices Keyboards Configuration

# User Testing

For this, we want to make sure that updating the Sentry component does not break existing functionality anywhere in our code.

* **TEST_DEVELOPER_SERVER**: Load Keyman Developer, start a web test for a keyboard, and verify that the web test functions normally (in a web browser). Press F12 to verify that no errors have been reported in the Web Developer Console
* **TEST_DEVELOPER_EDITORS**: Load Keyman Developer, open a keyboard source, and verify that the text editor and touch layout editor continue to function normally. Press F12 to verify that no errors have been reported in the Developer Console.
* **TEST_DESKTOP_CONFIG**: Start Keyman Configuration for Windows, verify that it continues to operate normally. Press F12 to verify that no errors have been reported in the Developer Console.

### CODE REVIEW

**Comment by mcdurdin** [comment]
> Starts from the parent folder, as it is possible that paths starting with .. may be ignored, as they were not being uploaded to sentry.

I was wrong, I missed them in the log. The sourcemaps were uploading previously:

From 17.0.273 release [build](https://build.palaso.org/buildConfiguration/Keyman_Developer_Release/444654?buildTab=log&focusLine=16779&logView=flowAware&linesState=313.14245) (before this PR is merged):
```
02:17:38       ~/build/src/site/resource/keymanweb.es5.js (sourcemap at keymanweb.es5.js.map)
02:17:38       ~/build/src/site/resource/keymanweb.js (sourcemap at keymanweb.js.map)
02:17:38       ~/build/src/site/resource/kmwuibutton.js (sourcemap at kmwuibutton.js.map)
02:17:38       ~/build/src/site/resource/kmwuifloat.js (sourcemap at kmwuifloat.js.map)
02:17:38       ~/build/src/site/resource/kmwuitoggle.js (sourcemap at kmwuitoggle.js.map)
02:17:38       ~/build/src/site/resource/kmwuitoolbar.js (sourcemap at kmwuitoolbar.js.map)
```

So the issue may be that we need to rewrite the sourcemap references with a different path? Looking at a variety of crash reports, some seem to have sourcemaps:
* https://keyman.sentry.io/issues/4561857038/

But others don't:
* https://keyman.sentry.io/issues/4972536418/ (original report for this PR)
* https://keyman.sentry.io/issues/4715313714/

@jahorton any ideas?

(This PR isn't worthless because it cleans up the upload a little, but it isn't really doing what we need.)

**Comment by mcdurdin** [comment]
I have checked the build logs and the 413 and 414 errors are resolved with the `--no-dedupe` flag. And it's actually faster anyway hah by 10 seconds or so. Good network connection probably helps.

**Comment by jahorton** [comment]
> > Starts from the parent folder, as it is possible that paths starting with .. may be ignored, as they were not being uploaded to sentry.
> 
> I was wrong, I missed them in the log. The sourcemaps were uploading previously:
> 
> From 17.0.273 release [build](https://build.palaso.org/buildConfiguration/Keyman_Developer_Release/444654?buildTab=log&focusLine=16779&logView=flowAware&linesState=313.14245) (before this PR is merged):
> 
> ```
> 02:17:38       ~/build/src/site/resource/keymanweb.es5.js (sourcemap at keymanweb.es5.js.map)
> 02:17:38       ~/build/src/site/resource/keymanweb.js (sourcemap at keymanweb.js.map)
> 02:17:38       ~/build/src/site/resource/kmwuibutton.js (sourcemap at kmwuibutton.js.map)
> 02:17:38       ~/build/src/site/resource/kmwuifloat.js (sourcemap at kmwuifloat.js.map)
> 02:17:38       ~/build/src/site/resource/kmwuitoggle.js (sourcemap at kmwuitoggle.js.map)
> 02:17:38       ~/build/src/site/resource/kmwuitoolbar.js (sourcemap at kmwuitoolbar.js.map)
> ```
> 
> So the issue may be that we need to rewrite the sourcemap references with a different path? Looking at a variety of crash reports, some seem to have sourcemaps:
> 
>     * https://keyman.sentry.io/issues/4561857038/
> 
> 
> But others don't:
> 
>     * https://keyman.sentry.io/issues/4972536418/ (original report for this PR)
> 
>     * https://keyman.sentry.io/issues/4715313714/
> 
> 
> @jahorton any ideas?
> 
> (This PR isn't worthless because it cleans up the upload a little, but it isn't really doing what we need.)

Yeah, I've been noticing a similar behavior from Sentry events for Android recently.  That's... a pretty good question, and I don't have a solid answer yet.  I suspect it's something to do with paths... except that it works _sometimes_ and not other times, without a currently distinguishable reason.

For Android and iOS, our Sentry integration code does a little internal path-rewriting before marshalling events to the server; it's supposed to help alleviate pathing issues.  But, as noted, something's been going funny lately at times for events sent from Android.

**Review by darcywong00 — APPROVED** [review]
lgtm

File: windows/src/desktop/kmshell/xml/sentry.bundle.min.js [hunk 1]
```diff
@@ -1,3 +1,3 @@
...
+/*! @sentry/browser 7.102.0 (4c0fa37) | https://github.com/getsentry/sentry-javascript */
```
Comment (darcywong00) [inline windows/src/desktop/kmshell/xml/sentry.bundle.min.js:1]: > Updates the sentry-browser SDK to 7.102.0 on the following components:
> Keyman Configuration

Does this mean the scope of the PR includes windows? (for version-history)

**Comment by bharanidharanj** [comment]
# Test Results

* **TEST_DEVELOPER_SERVER (PASSED):** Tested with the attached PR build (Keyman Developer 17.0.273-beta-test-10798) on Windows 11 OS (VM) and here is my observation : 1. Opened an exisitng project (sil_cameroon_qwerty) using Keyman Developer. 2. Compiled it. 3. Clicked the Build tab. Opened the local host browser (in Mozilla firefox) . 4. Selected the sil_cameroon_qwerty keyboard. 5. Pressed F12 key. 6. Noticed that it is showing some error message (?) on the console window. 7. Tried to open the same keyboard on Chrome browser and it doesn't show up any error message.  

..in Mozilla browser

   


..in Chrome browser

**Comment by bharanidharanj** [comment]
# Test Results

* **TEST_DEVELOPER_EDITORS (FAILED):** Tested with the attached PR build (Keyman Developer 17.0.273-beta-test-10798) on Windows 11 OS (VM) and here is my observation : 1. Opened an existing project (sil_cameroon_qwerty) using Keyman Developer. 2. Compiled it. 3. Clicked the Touch Layout tab. 4. Pressed F12 in the Touch Layout editor shows an error message in the Console window.

**Comment by bharanidharanj** [comment]
# Test Results

* **TEST_DESKTOP_CONFIG (FAILED):** Tested with the attached PR build (Keyman Developer 17.0.273-beta-test-10798) on Windows 11 OS (VM) and here is my observation: 1. Opened an existing project (sil_cameroon_qwerty) using Keyman Developer. 2. Compiled it. 3. Clicked the Build tab. 4. Clicked the Install keyboard button. 5. Verified that the Keyboard was installed and appeared in the Keyman Configuration window. 6. Pressed F12 key. 7. Noticed that an error message appeared on the Console window.

**Comment by mcdurdin** [comment]
Thank you @bharanidharanj for running those tests. It turns out that none of those warnings and errors in the Developer console are important, and are not related to the changes here, so I am going to go ahead and mark the tests as passed.

TEST_DEVELOPER_EDITORS PASSED
TEST_DESKTOP_CONFIG PASSED

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.275-beta](https://keyman.com/downloads/releases/beta/17.0.275)

**Pull request merged**